eth/tracers/internal/tracetest/util.go

```diff
@@ -46,6 +46,7 @@ func (c *callContext) toBlockContext(genesis *core.Genesis) vm.BlockContext {
 	}
 	if genesis.Config.IsLondon(context.BlockNumber) {
 		context.BaseFee = (*big.Int)(c.BaseFee)
+		context.Random = &genesis.Mixhash
```

Comment:
Random is introduced in Merge, not London

Reply:
Can you give a suggestion change?

I'm not sure how to update it.

should I use isShanghai?

Reply:
or TerminalTotalDifficultyPassed?

Reply:
You can check the difficulty. The difficulty will be zero since the Merge Fork

e.g. 

```go
if header.Difficulty.Sign() == 0 {
		random = &header.MixDigest
	}
```